cachi2/core/resolver.py

```diff
@@ -22,7 +27,39 @@
 supported_package_managers = list(_package_managers)
 
 
-def resolve_packages(request: Request) -> RequestOutput:
+@contextmanager
+def back_up_request_source(request: Request, backup_path: Path) -> Iterator:
+    """Switches the source_dir of a Request for a full copy."""
+    original_source_dir = request.source_dir
+    source_backup = copy_directory(original_source_dir.path, backup_path)
+    request.source_dir = RootedPath(source_backup)
+
+    try:
+        yield request
+    finally:
+        if backup_path.exists():
+            shutil.rmtree(backup_path)
```

Comment:
Hmm, I'm not sure it `path.exists` is a sufficient check to perform the removal afterwards safely (it probably is for our use case), well, at least not without `ignore_errors=True`, but in context of temporary directories I mentioned in my other comments, this would likely become a `dir.cleanup(ignore_cleanup_errors=True)` which is essentially the same thing wrapped by different API.